### PR TITLE
fix(security) hide password value

### PR DESCRIPTION
**Pull request by loiclau**
## Description

hide password value with inspector

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 19.04.x
- [x] 19.10.x
- [x] 20.04.x
- [x] 20.10.x (master)

<h2> How this pull request can be tested ? </h2>

inspect input

## Checklist

- [x] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/Administration/parameters/ldap/form.php [hunk 1]
```diff
@@ -291,6 +291,7 @@
 $cloneSet[] = $cbtls;
 
 $gopt = array_merge($defaultOpt, $gopt);
+
```
Comment (loiclau) [inline www/include/Administration/parameters/ldap/form.php:294]:
```suggestion
```

File: www/class/centreonLDAP.class.php [hunk 2]
```diff
@@ -1174,6 +1177,7 @@ public function setGeneralOptions($arId, $options)
                          VALUES (" . $this->db->escape($arId) . ", '" . $this->db->escape($key) . "', '" .
                          $this->db->escape($value, false) . "')";
             }
+            var_dump($query);
```
Comment (kduret) [inline www/class/centreonLDAP.class.php:1180]: remove debug

File: www/class/centreonLDAP.class.php [hunk 3]
```diff
@@ -1111,6 +1111,10 @@ protected function updateLdapServers($arId)
     public function setGeneralOptions($arId, $options)
     {
         $gopt = $this->getGeneralOptions($arId);
+
+        if (isset($gopt["bind_pass"]) && $gopt["bind_pass"] === "******") {
```
Comment (kduret) [inline www/class/centreonLDAP.class.php:1115]: `******` should be a constant to avoid typo issue when it's used

**Review by kduret — CHANGES_REQUESTED** [review]

File: www/include/Administration/parameters/knowledgeBase/formKnowledgeBase.php [hunk 4]
```diff
@@ -33,14 +33,19 @@
  *
  */
 
+const PWS_OCCULTATION = '******';
```
Comment (kduret) [inline www/include/Administration/parameters/knowledgeBase/formKnowledgeBase.php:36]: you'll probably have a notice cause this const is already defined in `www/include/Administration/parameters/general/form.php`

**Review by kduret — CHANGES_REQUESTED** [review]
Regarding current legacy architecture, you should declare `public const PASSWD_OCCULTATION = '******';` in CentreonAuth.class.php
You could reuse it everywhere by using : `CentreonAuth::PASSWD_OCCULTATION`
So it will be define in a single place and use the power of constants ;)

**Review by kduret — DISMISSED** [review]

File: www/class/centreonLDAP.class.php [hunk 5]
```diff
@@ -1193,13 +1198,13 @@ public function setGeneralOptions($arId, $options)
     public function getGeneralOptions($arId)
     {
         $gopt = array();
-        $query = "SELECT `ari_name`, `ari_value` 
-                 FROM `auth_ressource_info`
-                 WHERE ar_id = " . $this->db->escape($arId);
+        $query = "SELECT `ari_name`, `ari_value` FROM `auth_ressource_info` 
+WHERE `ari_name` <> 'bind_pass' AND ar_id = " . $this->db->escape($arId);
```
Comment (jeremyjaouen) [inline www/class/centreonLDAP.class.php:1202]: bad indent

**Review by jeremyjaouen — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**